File: Longest Consecutive Sequence.java
public class Solution {
    public int longestConsecutive(int[] num) {
        HashSet<Integer> cache = new HashSet<Integer>();
        for (int i = 0; i < num.length; i++) {
            cache.add(num[i]);
        }
        HashSet<Integer> checked = new HashSet<Integer>();
        int max_len = 0;
        for (int i = 0; i < num.length; i++) {
            if (checked.contains(num[i]))
                continue;
            int len = 1;
            int n = num[i] + 1;
            while (cache.contains(n)) {
                checked.add(n);
                n++;
                len++;
            }
            n = num[i] - 1;
            while (cache.contains(n)) {
                checked.add(n);
                n--;
                len++;
            }
            if (len > max_len)
                max_len = len;
        }
        return max_len;
    }
}

File: Longest Consecutive Sequence.py
from sets import Set
class Solution:
	# @param num, a list of integer
	# @return an integer
	def longestConsecutive(self, num):
		cache = Set(num)
		checked = Set()
		max_length = 0
		for n in num:
			if (n in checked):
				continue
			length = 1
			p = n + 1
			while (p in cache):
				checked.add(p)
				p += 1
				length += 1
			m = n - 1
			while (m in cache):
				checked.add(m)
				m -= 1
				length += 1
			if (length > max_length):
				max_length = length
		return max_lengthmax_length